nomos-services/api/src/http/storage.rs

```diff
@@ -96,3 +109,67 @@ where
         .transpose()
         .map_err(super::DynError::from)
 }
+
+pub async fn get_shares<StorageOp, DaBlob>(
+    handle: &overwatch::overwatch::handle::OverwatchHandle,
+    blob_id: DaBlob::BlobId,
+    requested_shares: Vec<DaBlob::ColumnIndex>,
+    filter_shares: Vec<DaBlob::ColumnIndex>,
+    return_available: bool,
+) -> Result<impl Stream<Item = Result<Bytes, DynError>> + Send + Sync + 'static, DynError>
+where
+    StorageOp: StorageSerde + Send + Sync + 'static,
+    DaBlob: Blob,
+    DaBlob::BlobId: AsRef<[u8]> + DeserializeOwned + Send + Sync + 'static,
+    DaBlob::ColumnIndex: DeserializeOwned + Hash + Eq + Send + Sync + 'static,
+    DaBlob::LightBlob: LightBlob<ColumnIndex = <DaBlob as Blob>::ColumnIndex>
+        + Serialize
+        + DeserializeOwned
+        + Send
+        + Sync
+        + 'static,
+    StorageOp::Error: Send + Sync,
+{
+    let storage_relay = handle
+        .relay::<StorageService<RocksBackend<StorageOp>>>()
+        .connect()
+        .await?;
+
+    let shares_prefix = key_bytes(DA_BLOB_PREFIX, blob_id.as_ref());
```

Comment:
Wait, how are we storing shares? We should be able to ask for a `blob_id` + `share_idx`.